### PR TITLE
chore: migrate example to `react-native-test-app`

### PR DESCRIPTION
**Pull request by szymonrybczak**
# Summary

In this Pull Request I've migrated Example project to [`react-native-test-app`](https://github.com/microsoft/react-native-test-app).

| | Old arch | New arch (Bridgeless) |
|--------|--------|--------|
| Android | CleanShot 2024-06-19 at 10 42 05@2x | CleanShot 2024-06-19 at 10 31 45@2x |
| iOS | CleanShot 2024-06-19 at 10 46 24@2x |  CleanShot 2024-06-19 at 10 32 11@2x |

## Test Plan

App should properly run and build on all platforms that `react-native-pager-view` supports.

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅   |
| Android |    ✅     |
| visionOS |    ✅     |

## Checklist

- [x] I have tested this on a device and a simulator
- [ ] I added the documentation in `README.md`
- [ ] I updated the typed files (TS and Flow)

### CODE REVIEW

**Comment by troZee** [comment]
@szymonrybczak could you rebase your PR? 🙏

File: package.json [hunk 1]
```diff
@@ -66,7 +66,7 @@
     "eslint-plugin-ft-flow": "^3.0.1",
     "eslint-plugin-prettier": "^4.0.0",
     "jest": "^28.1.1",
-    "metro-react-native-babel-preset": "0.76.8",
+    "metro-react-native-babel-preset": "^0.77.0",
```
Comment (thymikee) [inline package.json:69]: let's make sure these dev deps are necessary and if true, make them the same versions as in example app

Comment (szymonrybczak) [reply package.json:68]: https://github.com/callstack/react-native-pager-view/pull/834/commits/0a7534637f52e7d557be7852381de77514990f79

**Comment by szymonrybczak** [comment]
CI is failing for iOS E2E tests because app built inside Release mode cannot grab `.jsbundle`, I tried also in a fresh project and it looks like something is broken:
CleanShot 2024-07-01 at 15 12 41@2x

```
npx --package react-native-test-app@latest init
yarn build:ios
yarn ios --mode "Release"
```

cc: @tido64 any ideas? Is there maybe anything that I'm missing in the config for app to run properly?

**Comment by tido64** [comment]
> cc: @tido64 any ideas? Is there maybe anything that I'm missing in the config for app to run properly?

When is `pod install` run? It needs to be run _after_ bundling so that CocoaPods can create an entry for it in the generated Xcode project files. You can check if it's present if you open `.xcworkspace` in Xcode.

**Comment by szymonrybczak** [comment]
@tido64 I did give it [a try](https://github.com/callstack/react-native-pager-view/pull/834/commits/0fb2f758f8c68b488b5c56565a15050b66a637ca), and it seems like the results is still the same :(

**Comment by tido64** [comment]
@szymonrybczak Do you have repro steps? Seems to work for me. This is what I ran:

```sh
yarn
cd example
yarn
yarn build:ios
pod install --project-directory=ios
yarn ios --no-packager --simulator 'iPhone 15 Pro'
```

image

File: example/package.json [hunk 2]
```diff
@@ -3,49 +3,53 @@
   "version": "0.0.1",
   "private": true,
   "scripts": {
-    "android": "react-native run-android",
-    "ios": "react-native run-ios --list-devices",
+    "android": "react-native run-android --appId com.pagerviewexample",
+    "build:android": "npm run mkdist && react-native bundle --entry-file index.js --platform android --dev true --bundle-output dist/main.android.jsbundle --assets-dest dist/res",
+    "build:ios": "npm run mkdist && react-native bundle --entry-file index.js --platform ios --dev true --bundle-output dist/main.ios.jsbundle --assets-dest dist",
```
Comment (szymonrybczak) [inline example/package.json:8]:
```suggestion
    "build:ios": "npm run mkdist && react-native bundle --entry-file index.js --platform ios --dev false --bundle-output dist/main.ios.jsbundle --assets-dest dist",
```

File: example/package.json [hunk 3]
```diff
@@ -3,49 +3,53 @@
   "version": "0.0.1",
   "private": true,
   "scripts": {
-    "android": "react-native run-android",
-    "ios": "react-native run-ios --list-devices",
+    "android": "react-native run-android --appId com.pagerviewexample",
+    "build:android": "npm run mkdist && react-native bundle --entry-file index.js --platform android --dev true --bundle-output dist/main.android.jsbundle --assets-dest dist/res",
+    "build:ios": "npm run mkdist && react-native bundle --entry-file index.js --platform ios --dev false --bundle-output dist/main.ios.jsbundle --assets-dest dist",
```
Comment (tido64) [inline example/package.json:8]: You might want to add `--dev false` in `.github/workflows/ios.yml` instead.

Comment (szymonrybczak) [reply example/package.json:8]: I think this script will be only used for building on CI, since for development `yarn ios` already fires up packager.

**Review by troZee — APPROVED** [review]

**Pull request merged**